### PR TITLE
MNG-2741 - Adds registrationLineItems and registrationLineItemOptions

### PR DESCRIPTION
**Pull request by Ollo**
<https://teamsnap.atlassian.net/browse/MNG-2741>

### CODE REVIEW

File: src/types.coffee [hunk 1]
```diff
@@ -58,6 +58,8 @@ types = [
   'paymentNote'
   'plan'
   'registrationForm'
+  'registrationLineItemOptions'
+  'registrationLineItems'
```
Comment (matthopson) [inline src/types.coffee]: These should be singular types:
```
registrationLineItemOption
registrationLineItem
```

**Comment by matthopson** [comment]
It may seem like re-arranging deck chairs on the Titanic, but we're also missing docs for these. I know we're going to move on beyond the SDK, but I'd like to keep it complete. If you're going to be adding additional methods for these collections, you can drop the docs in then (they're pretty simple).

File: src/collections/registrationLineItemOptions.coffee [hunk 2]
```diff
@@ -0,0 +1,6 @@
+exports.loadRegistrationLineItemOptions = (params, callback) ->
+  unless params and typeof params is 'object'
+    throw new TSArgsError 'teamsnap.registrationLineItemOptions',
+      'must provide query parameters'
+
+  @loadItems 'registrationLineItemOptions', params, callback
```
Comment (matthopson) [inline src/collections/registrationLineItemOptions.coffee]: Should be calling `loadItems` against the singular type.

File: src/collections/registrationLineItems.coffee [hunk 3]
```diff
@@ -0,0 +1,6 @@
+exports.loadRegistrationLineItems = (params, callback) ->
+  unless params and typeof params is 'object'
+    throw new TSArgsError 'teamsnap.registrationLineItems',
+      'must provide query parameters'
+
+  @loadItems 'registrationLineItems', params, callback
```
Comment (matthopson) [inline src/collections/registrationLineItems.coffee]: Same here - should be `@loadItems registrationLineItem`

File: src/collections/registrationLineItemOptions.coffee [hunk 4]
```diff
@@ -0,0 +1,6 @@
+exports.loadRegistrationLineItemOptions = (params, callback) ->
+  unless params and typeof params is 'object'
+    throw new TSArgsError 'teamsnap.registrationLineItemOptions',
```
Comment (matthopson) [inline src/collections/registrationLineItemOptions.coffee]: Should include the method name in the error:
`throw new TSArgsError 'teamsnap.loadRegistrationLineItemOptions'`

File: src/collections/registrationLineItems.coffee [hunk 5]
```diff
@@ -0,0 +1,6 @@
+exports.loadRegistrationLineItems = (params, callback) ->
+  unless params and typeof params is 'object'
+    throw new TSArgsError 'teamsnap.registrationLineItems',
```
Comment (matthopson) [inline src/collections/registrationLineItems.coffee]: method name here

File: src/collections/registrationLineItemOptions.coffee [hunk 6]
```diff
@@ -0,0 +1,6 @@
+exports.loadRegistrationLineItemOptions = (params, callback) ->
+  unless params and typeof params is 'object'
```
Comment (matthopson) [inline src/collections/registrationLineItemOptions.coffee]: There may be a few cases where we've diverted from this - but we should allow a `teamId` in addition to an object with params. That would look more like:
```javascript
exports.loadRegistrationLineItemOptions = (params, callback) ->
  if @isId params
    params = teamId: params
  else unless params and typeof params is 'object'
    throw new TSArgsError 'teamsnap.loadRegistrationLineItemOptions', 'must provide a teamId or
      query parameters'

@loadItems 'registrationLineItemOption', params, callback
```

Comment (matthopson) [inline src/collections/registrationLineItemOptions.coffee]: This allows you to simply do: `loadRegistrationLineItemOptions(teamId)` to load for a given team.

Comment (dmccraw) [inline src/collections/registrationLineItemOptions.coffee]: Matt, there isn't any connection to a team for a RegistrationLIneItem or a RegistrationLineItemOption. These are for registrants going through registration before they are ever assigned to a team. There is a division associated with them.

Comment (matthopson) [inline src/collections/registrationLineItemOptions.coffee]: Ah. Good thing I provisioned with "There may be a few cases where we've diverted from this"

Thanks Dustin!

**Comment by Ollo** [comment]
@matthopson updated those to remove the `teamId` check

**Pull request merged**